### PR TITLE
Add rose75 keyboard

### PR DESCRIPTION
**Pull request by SmollChungus**
## Description

Addition of firmware for rose75 PCB

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/rose75/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	),
+
+	[1] = LAYOUT(
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+	),
```
Comment (zvecr) [inline keyboards/rose75/keymaps/via/keymap.c:23]:
```suggestion
	)
```

File: keyboards/rose75/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	),
+
+	[1] = LAYOUT(
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+	),
```
Comment (zvecr) [inline keyboards/rose75/keymaps/default/keymap.c:24]:
```suggestion
	)
```

File: keyboards/rose75/info.json [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    },
+    "ws2812": {
+        "pin": "GP29",
+        "driver": "vendor"
+    },
+    "rgb_matrix": {
+        "driver": "ws2812",
+        "layout": [
+            { "flags": 8, "matrix": [ 3, 0 ],  "x": 0,   "y": 0 },
+            { "flags": 8, "matrix": [ 3, 0 ],  "x": 15,  "y": 15 }
```
Comment (zvecr) [inline keyboards/rose75/info.json:34]: You should not have duplicate matrix items here

File: keyboards/rose75/config.h
Comment (zvecr) [inline keyboards/rose75/config.h:1]: This file should be removed

File: keyboards/rose75/info.json [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        ]
```
Comment (zvecr) [inline keyboards/rose75/info.json:35]:
```suggestion
        ],
        "max_brightness": 210,
        "sleep": true
```

File: keyboards/rose75/info.json [hunk 5]
```diff
@@ -0,0 +1,125 @@
+{
+    "manufacturer": "Matthijs Muller",
+    "keyboard_name": "ROSE75",
+    "maintainer": "Matthijs Muller",
+    "processor": "RP2040",
+    "bootloader": "rp2040",
+    "usb": {
+        "vid": "0x5363",
+        "pid": "0x7503",
+        "device_version": "0.0.1"
```
Comment (zvecr) [inline keyboards/rose75/info.json:10]:
```suggestion
        "device_version": "0.0.1",
        "force_nkro": true
```

File: keyboards/rose75/readme.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# rose75
+
+![rose75](https://i.imgur.com/f64Ttib.png)
+
+
+* Keyboard Maintainer: [Matthijs Muller](https://github.com/Smollchungus)
+* Hardware Supported: Rose75 Keyboard
+
```
Comment (zvecr) [inline keyboards/rose75/readme.md:8]:
```suggestion
* Keyboard Maintainer: [Matthijs Muller](https://github.com/Smollchungus)
* Hardware Supported: Rose75 Keyboard
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/rose75/info.json (hunk 3 above)
Comment (SmollChungus) [reply keyboards/rose75/info.json]: My bad, appreciate the review! As they are just 2 indicators, I labeled them as 0,0, and 0,1 for now.

File: keyboards/rose75/config.h [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+// RGB indicators
+
+#define WS2812_PIO_USE_PPIO1
+#define RGB_MATRIX_LED_COUNT 2
+#define RGB_DISABLE_WHEN_USB_SUSPENDED
+#define RGB_MATRIX_MAXIMUM_BRIGHTNESS 210
+
+//Force NKRO on startup
+#define FORCE_NKRO
```
Comment (waffle87) [inline keyboards/rose75/config.h:14]: This is already specified in `info.json`.

File: keyboards/rose75/config.h [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+// RGB indicators
+
+#define WS2812_PIO_USE_PPIO1
+#define RGB_MATRIX_LED_COUNT 2
+#define RGB_DISABLE_WHEN_USB_SUSPENDED
+#define RGB_MATRIX_MAXIMUM_BRIGHTNESS 210
```
Comment (waffle87) [inline keyboards/rose75/config.h:11]: These definitions are all already specified in `info.json` (`WS2812_PIO_USE_PPIO1` is invalid and does not exist)

File: keyboards/rose75/config.h
Comment (waffle87) [inline keyboards/rose75/config.h:1]: This file should just be removed with the above comments.

File: keyboards/rose75/readme.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
+# rose75
+
+![rose75](https://i.imgur.com/f64Ttib.png)
```
Comment (waffle87) [inline keyboards/rose75/readme.md:3]:
```suggestion
![rose75](https://i.imgur.com/f64Ttibh.png)
```

File: keyboards/rose75/rose75.c [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+    }
+    if (host_keyboard_led_state().caps_lock) {
+        rgb_matrix_set_color(0, 200, 0, 0);
+        rgb_matrix_set_color(1, 200, 0, 0);
+    } else {
+        rgb_matrix_set_color(0, 0, 0, 0);
+        rgb_matrix_set_color(1, 0, 0, 0);
+    }
+    return true;
+}
```
Comment (waffle87) [inline keyboards/rose75/rose75.c:18]:
```suggestion

#ifdef RGB_MATRIX_ENABLE
bool rgb_matrix_indicators_kb(void) {
    if (!rgb_matrix_indicators_user()) {
         return false;
    }
    if (host_keyboard_led_state().caps_lock) {
        rgb_matrix_set_color(0, 200, 0, 0);
        rgb_matrix_set_color(1, 200, 0, 0);
    } else {
        rgb_matrix_set_color(0, 0, 0, 0);
        rgb_matrix_set_color(1, 0, 0, 0);
    }
    return true;
}
#endif
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by SmollChungus** [comment]
thanks for checking! removed the config.h from the pr.

**Comment by SmollChungus** [comment]
Just noticed I mistakenly targeted QMK:master branch instead of develop, and I can't see a way to change it inside this PR. I will close this PR and target the correct branch in the new PR. Apologies (:

**Comment by sigprof** [comment]
> Just noticed I mistakenly targeted QMK:master branch instead of develop

Does the code for this keyboard actually depend on some core code that is only in the `develop` branch?

Adding new keyboards on the `master` branch is fine.  The cases when you need to target the `develop` branch are:

- Renaming or moving some previously existing keyboards (this includes the case when you are introducing a `rev2` for some board, and need to move the previously existing code into the `rev1` subfolder).
- Doing substantial changes to some existing keyboard even without renaming it (but small bugfixes like fixing some wrong pin or matrix location for some key may be fine to do on the `master` branch directly).
- Changing core code (adding new features or modifying existing ones); there might be exceptions for small and obvious bugfixes.
- Adding a new keyboard which uses some new core feature that is currently available only in the `develop` branch.

There is some chance that your code fits into the last bullet point, but I don't see anything obvious; if the keyboard code actually works fine in the `master` branch, you may just target `master`.


> and I can't see a way to change it inside this PR

The GitHub UI for changing the target branch of a PR is not obvious. First, you need to click the Edit button near the top of the page:
image
Then, if the PR is open, the text that shows the target branch changes into a dropdown (here I show an example from another PR which is still open):
image

But if you actually don't need to use the `develop` branch, you may just reopen this PR again (if you have difficulties with doing that, write a comment here).

File: keyboards/rose75/keymaps/via/keymap.c [hunk 11]
```diff
@@ -0,0 +1,15 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
```
Comment (zvecr) [inline keyboards/rose75/keymaps/via/keymap.c:1]:
```suggestion
// Copyright 2023 Matthijs Muller (@Smollchungus)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/rose75/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT(
+		KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,            KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_END,
+		KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_HOME,
+		KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_PGUP,
+		KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,           KC_PGDN,
+		KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,   KC_END,
+		KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, KC_RCTL,          KC_LEFT, KC_DOWN, KC_RGHT
+	)
```
Comment (drashna) [inline keyboards/rose75/keymaps/default/keymap.c:15]:
```suggestion
    [0] = LAYOUT(
        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,            KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_END,
        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_HOME,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_PGUP,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,           KC_PGDN,
        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,   KC_END,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, KC_RCTL,          KC_LEFT, KC_DOWN, KC_RGHT
    )
```

File: keyboards/rose75/keymaps/via/keymap.c [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT(
+		KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,            KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_END,
+		KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_HOME,
+		KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_PGUP,
+		KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,           KC_PGDN,
+		KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,   KC_END,
+		KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, KC_RCTL,          KC_LEFT, KC_DOWN, KC_RGHT
+	)
```
Comment (drashna) [inline keyboards/rose75/keymaps/via/keymap.c:15]:
```suggestion
    [0] = LAYOUT(
        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,            KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_END,
        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_HOME,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_PGUP,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,           KC_PGDN,
        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,   KC_END,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, KC_RCTL,          KC_LEFT, KC_DOWN, KC_RGHT
    )
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/rose75/info.json
Comment (fauxpark) [inline keyboards/rose75/info.json:1]: Please run `qmk format-json -i` on this file.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/rose75/info.json
Comment (SmollChungus) [reply keyboards/rose75/info.json:1]: Done, my apologies

File: keyboards/rose75/readme.md [hunk 14]
```diff
@@ -0,0 +1,24 @@
+# rose75
+
+![rose75](https://i.imgur.com/f64Ttibh.png)
```
Comment (waffle87) [inline keyboards/rose75/readme.md:3]:
```suggestion
![rose75](https://i.imgur.com/f64Ttib.png)
```

**Review by waffle87 — APPROVED** [review]

**Pull request merged**